libs/libGamedata/lua/CampaignDataLoader.cpp

```diff
@@ -28,8 +28,10 @@ bool CampaignDataLoader::Load()
     auto curFile_ = basePath_ / "campaign.lua";
     try
     {
-        if(!loadScript(curFile_))
+        if(!loadScript(curFile_) || lua["campaign"].isNilref() || lua["campaign"].type() != LUA_TTABLE)
```

Comment:
Maybe just get the campaign as a lua ref once instead of 3 times? See e.g. https://github.com/Return-To-The-Roots/s25client/blob/29dc07bfb32d96c0e931ddffa5d4d4a39419549c/libs/libGamedata/lua/LuaInterfaceBase.cpp#L109-L110 and IIRC `type() == LUA_TTABLE` implies `!isNilref()`
You could even throw an exception (caught below) if the entry does not exist to log that `campaign` is missing

Reply:
Changed